## Load Data

In [1]:
library(readr)
Sys.setenv("VROOM_CONNECTION_SIZE" = 131072 * 5)
gtex_v8_features <- read_csv("gtex_v8_features.csv")
gtex_v8_features <- as.data.frame(gtex_v8_features)

Rows: 4341 Columns: 3572
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr    (1): Name
dbl (3571): GTEX-1117F-3226-SM-5N9CT, GTEX-111FC-3126-SM-5GZZ2, GTEX-111FC-3...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [2]:
if (any(duplicated(gtex_v8_features[,1]))) {
  # If duplicates exist, make them unique (e.g., by adding a suffix)
  gtex_v8_features$Name <- make.unique(gtex_v8_features$Name)
}

# Now assign row names
rownames(gtex_v8_features) <- gtex_v8_features$Name
gtex_v8_features <- gtex_v8_features[,-1]

In [3]:
head(gtex_v8_features)

GTEX-1117F-3226-SM-5N9CT GTEX-111FC-3126-SM-5GZZ2
ENSG00000168137.15 18.39                    28.73                   
ENSG00000144550.12 32.46                    41.45                   
ENSG00000134072.10 44.12                    74.58                   
ENSG00000171148.13 49.81                    59.31                   
ENSG00000241553.12 65.27                    73.35                   
ENSG00000163703.17 47.54                    69.01                   
                   GTEX-111FC-3326-SM-5GZYV GTEX-111YS-0006-SM-5NQBE
ENSG00000168137.15  75.19                    10.37000               
ENSG00000144550.12  74.53                     0.03452               
ENSG00000134072.10  56.41                    45.52000               
ENSG00000171148.13  44.51                    96.17000               
ENSG00000241553.12  58.07                   352.50000               
ENSG00000163703.17 116.70                     4.28000               
                   GTEX-1122O-0003-SM-5Q5DL GTEX-1122O-0005-SM-5O99J
ENSG00000168137.15  56.24000                 21.6400                
ENSG00000144550.12   0.03166                  0.1198                
ENSG00000134072.10   3.50400                 19.1800                
ENSG00000171148.13  54.53000                121.4000                
ENSG00000241553.12 221.80000                402.5000                
ENSG00000163703.17  14.92000                  3.8630                
                   GTEX-1128S-0005-SM-5P9HI GTEX-1128S-2726-SM-5H12C
ENSG00000168137.15  37.2400                 18.64                   
ENSG00000144550.12   0.4358                 31.25                   
ENSG00000134072.10  22.9900                 76.41                   
ENSG00000171148.13  37.0800                 46.69                   
ENSG00000241553.12 121.1000                 69.54                   
ENSG00000163703.17  14.4900                 56.51                   
                   GTEX-1128S-2826-SM-5N9DI GTEX-113IC-0006-SM-5NQ9C ⋯
ENSG00000168137.15  54.86                    24.2900                 ⋯
ENSG00000144550.12  62.51                     0.6124                 ⋯
ENSG00000134072.10  49.67                    82.1600                 ⋯
ENSG00000171148.13  35.51                   127.9000                 ⋯
ENSG00000241553.12  57.89                   355.0000                 ⋯
ENSG00000163703.17 135.40                    17.2600                 ⋯
                   GTEX-ZXG5-0011-R5b-SM-57WBN GTEX-ZXG5-0011-R6a-SM-57WBZ
ENSG00000168137.15 13.110                      15.810                     
ENSG00000144550.12  2.469                       1.503                     
ENSG00000134072.10 43.660                      68.150                     
ENSG00000171148.13 33.670                      37.420                     
ENSG00000241553.12 36.250                      52.850                     
ENSG00000163703.17 27.530                      41.200                     
                   GTEX-ZXG5-0011-R7b-SM-57WCC GTEX-ZXG5-0011-R8a-SM-57WD1
ENSG00000168137.15  9.279                       9.097                     
ENSG00000144550.12  1.231                       4.109                     
ENSG00000134072.10 19.550                      21.010                     
ENSG00000171148.13 27.720                      25.480                     
ENSG00000241553.12 21.100                      31.150                     
ENSG00000163703.17 13.740                      28.240                     
                   GTEX-ZYFD-2926-SM-5GID9 GTEX-ZYFD-3026-SM-5E44C
ENSG00000168137.15  66.67                  22.40                  
ENSG00000144550.12  60.33                  50.95                  
ENSG00000134072.10  54.59                  97.18                  
ENSG00000171148.13  37.43                  63.70                  
ENSG00000241553.12  49.35                  85.22                  
ENSG00000163703.17 104.40                  63.50                  
                   GTEX-ZYY3-3026-SM-5GIEJ GTEX-ZYY3-3126-SM-5SI9L
ENSG

In [4]:
gtex_v8_target <- read_csv("gtex_v8_target.csv")
head(gtex_v8_target)

Rows: 6806 Columns: 63
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (14): SAMPID, SMCENTER, SMPTHNTS, SMTS, SMTSD, SMUBRID, SMNABTCH, SMNABT...
dbl (41): SMATSSCR, SMRIN, SMTSISCH, SMTSPAX, SME2MPRT, SMCHMPRS, SMNTRART, ...
lgl  (8): SMNUMGPS, SM550NRM, SM350NRM, SMMNCPB, SMMNCV, SMCGLGTH, SMGAPPCT,...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


SAMPID                        SMATSSCR SMCENTER SMPTHNTS SMRIN SMTS 
1 GTEX-1117F-0003-SM-58Q7G      NA       B1       NA        NA   Blood
2 GTEX-1117F-0003-SM-5DWSB      NA       B1       NA        NA   Blood
3 GTEX-1117F-0003-SM-6WBT7      NA       B1       NA        NA   Blood
4 GTEX-1117F-0011-R10a-SM-AHZ7F NA       B1, A1   NA        NA   Brain
5 GTEX-1117F-0011-R10b-SM-CYKQ8 NA       B1, A1   NA       7.2   Brain
6 GTEX-1117F-3226-SM-5N9CT       1       B1       2 pieces 6.2   Brain
  SMTSD                        SMUBRID SMTSISCH SMTSPAX ⋯ SME1ANTI SMSPLTRD
1 Whole Blood                  0013756 1188      NA     ⋯       NA      NA 
2 Whole Blood                  0013756 1188      NA     ⋯       NA      NA 
3 Whole Blood                  0013756 1188      NA     ⋯       NA      NA 
4 Brain - Frontal Cortex (BA9) 0009834 1193      NA     ⋯       NA      NA 
5 Brain - Frontal Cortex (BA9) 0009834 1193      NA     ⋯       NA      NA 
6 Brain - Cortex               0001870 1418     931     ⋯ 17609900 8163850 
  SMBSMMRT   SME1SNSE SME1PCTS SMRRNART  SME1MPRT SMNUM5CD SMDPMPRT SME2PCTS
1         NA       NA     NA          NA       NA NA       NA           NA  
2         NA       NA     NA          NA       NA NA       NA           NA  
3         NA       NA     NA          NA       NA NA       NA           NA  
4         NA       NA     NA          NA       NA NA       NA           NA  
5         NA       NA     NA          NA       NA NA       NA           NA  
6 0.00349458 17906900 50.418   0.0620363 0.993989 NA        0       50.191

In [5]:
gtex_v8_subject_phenotypes <- read_tsv("https://storage.googleapis.com/adult-gtex/annotations/v8/metadata-files/GTEx_Analysis_v8_Annotations_SubjectPhenotypesDS.txt")
head(gtex_v8_subject_phenotypes)

Rows: 980 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): SUBJID, AGE
dbl (2): SEX, DTHHRDY

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


SUBJID     SEX AGE   DTHHRDY
1 GTEX-1117F 2   60-69 4      
2 GTEX-111CU 1   50-59 0      
3 GTEX-111FC 1   60-69 1      
4 GTEX-111VG 1   60-69 3      
5 GTEX-111YS 1   60-69 0      
6 GTEX-1122O 2   60-69 0

### Merge Sample level Tissue information metadata with Subject level metadata

In [6]:
gtex_v8_target <- gtex_v8_target |>
  dplyr::mutate(SUBJID = stringr::str_extract(SAMPID, "^[^-]*-[^-]*"))
gtex_v8_target <- dplyr::left_join(gtex_v8_target, gtex_v8_subject_phenotypes, by = "SUBJID")
head(gtex_v8_target)

SAMPID                        SMATSSCR SMCENTER SMPTHNTS SMRIN SMTS 
1 GTEX-1117F-0003-SM-58Q7G      NA       B1       NA        NA   Blood
2 GTEX-1117F-0003-SM-5DWSB      NA       B1       NA        NA   Blood
3 GTEX-1117F-0003-SM-6WBT7      NA       B1       NA        NA   Blood
4 GTEX-1117F-0011-R10a-SM-AHZ7F NA       B1, A1   NA        NA   Brain
5 GTEX-1117F-0011-R10b-SM-CYKQ8 NA       B1, A1   NA       7.2   Brain
6 GTEX-1117F-3226-SM-5N9CT       1       B1       2 pieces 6.2   Brain
  SMTSD                        SMUBRID SMTSISCH SMTSPAX ⋯ SME1PCTS SMRRNART 
1 Whole Blood                  0013756 1188      NA     ⋯     NA          NA
2 Whole Blood                  0013756 1188      NA     ⋯     NA          NA
3 Whole Blood                  0013756 1188      NA     ⋯     NA          NA
4 Brain - Frontal Cortex (BA9) 0009834 1193      NA     ⋯     NA          NA
5 Brain - Frontal Cortex (BA9) 0009834 1193      NA     ⋯     NA          NA
6 Brain - Cortex               0001870 1418     931     ⋯ 50.418   0.0620363
  SME1MPRT SMNUM5CD SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY
1       NA NA       NA           NA   GTEX-1117F 2   60-69 4      
2       NA NA       NA           NA   GTEX-1117F 2   60-69 4      
3       NA NA       NA           NA   GTEX-1117F 2   60-69 4      
4       NA NA       NA           NA   GTEX-1117F 2   60-69 4      
5       NA NA       NA           NA   GTEX-1117F 2   60-69 4      
6 0.993989 NA        0       50.191   GTEX-1117F 2   60-69 4

### Retain only the intersection of the target dataframe and the feature dataframe

In [7]:
common_samples <- intersect(gtex_v8_target$SAMPID, colnames(gtex_v8_features))
print(length(common_samples))
Y <- as.data.frame(t(gtex_v8_features))
rownames(Y) <- colnames(gtex_v8_features)
colnames(Y) <- rownames(gtex_v8_features)
Y <- Y[rownames(Y) %in% common_samples, ]
head(Y)

[1] 3571


ENSG00000168137.15 ENSG00000144550.12
GTEX-1117F-3226-SM-5N9CT 18.39              32.46000          
GTEX-111FC-3126-SM-5GZZ2 28.73              41.45000          
GTEX-111FC-3326-SM-5GZYV 75.19              74.53000          
GTEX-111YS-0006-SM-5NQBE 10.37               0.03452          
GTEX-1122O-0003-SM-5Q5DL 56.24               0.03166          
GTEX-1122O-0005-SM-5O99J 21.64               0.11980          
                         ENSG00000134072.10 ENSG00000171148.13
GTEX-1117F-3226-SM-5N9CT 44.120              49.81            
GTEX-111FC-3126-SM-5GZZ2 74.580              59.31            
GTEX-111FC-3326-SM-5GZYV 56.410              44.51            
GTEX-111YS-0006-SM-5NQBE 45.520              96.17            
GTEX-1122O-0003-SM-5Q5DL  3.504              54.53            
GTEX-1122O-0005-SM-5O99J 19.180             121.40            
                         ENSG00000241553.12 ENSG00000163703.17
GTEX-1117F-3226-SM-5N9CT  65.27              47.540           
GTEX-111FC-3126-SM-5GZZ2  73.35              69.010           
GTEX-111FC-3326-SM-5GZYV  58.07             116.700           
GTEX-111YS-0006-SM-5NQBE 352.50               4.280           
GTEX-1122O-0003-SM-5Q5DL 221.80              14.920           
GTEX-1122O-0005-SM-5O99J 402.50               3.863           
                         ENSG00000254999.3 ENSG00000157020.17
GTEX-1117F-3226-SM-5N9CT 148.4              32.88            
GTEX-111FC-3126-SM-5GZZ2 174.3              34.30            
GTEX-111FC-3326-SM-5GZYV 165.3              50.47            
GTEX-111YS-0006-SM-5NQBE 164.0              36.02            
GTEX-1122O-0003-SM-5Q5DL 251.8             132.70            
GTEX-1122O-0005-SM-5O99J 135.9              32.77            
                         ENSG00000157087.17 ENSG00000132164.9 ⋯
GTEX-1117F-3226-SM-5N9CT  37.31000          13.830000         ⋯
GTEX-111FC-3126-SM-5GZZ2  86.51000          13.280000         ⋯
GTEX-111FC-3326-SM-5GZYV 170.40000           2.383000         ⋯
GTEX-111YS-0006-SM-5NQBE   0.01537           0.005864         ⋯
GTEX-1122O-0003-SM-5Q5DL   0.04229           0.021520         ⋯
GTEX-1122O-0005-SM-5O99J   0.05334           0.048850         ⋯
                         ENSG00000155368.16 ENSG00000144118.13
GTEX-1117F-3226-SM-5N9CT 105.40              29.50            
GTEX-111FC-3126-SM-5GZZ2 120.70              55.31            
GTEX-111FC-3326-SM-5GZYV 170.30              25.54            
GTEX-111YS-0006-SM-5NQBE  21.86             231.70            
GTEX-1122O-0003-SM-5Q5DL 426.60              95.34            
GTEX-1122O-0005-SM-5O99J  11.14             513.70            
                         ENSG00000136732.14 ENSG00000136717.14
GTEX-1117F-3226-SM-5N9CT   2.226            190.60            
GTEX-111FC-3126-SM-5GZZ2   1.793            228.80            
GTEX-111FC-3326-SM-5GZYV   2.749            168.50            
GTEX-111YS-0006-SM-5NQBE 123.800             15.07            
GTEX-1122O-0003-SM-5Q5DL  39.290             35.22            
GTEX-1122O-0005-SM-5O99J 220.900             10.70            
                         ENSG00000163161.12 ENSG00000136720.6
GTEX-1117F-3226-SM-5N9CT 43.44              39.460           
GTEX-111FC-3126-SM-5GZZ2 49.32              52.100           
GTEX-111FC-3326-SM-5GZYV 81.40              41.010           
GTEX-111YS-0006-SM-5NQBE 11.58               3.872           
GTEX-1122O-0003-SM-5Q5DL 58.23               5.941           
GTEX-1122O-0005-SM-5O99J 15.68               4.074           
                         ENSG00000152082.13 ENSG00000136710.9 ENSG00000136718.9
GTEX-1117F-3226-SM-5N9CT 186.80             38.43              37.43           
GTEX-111FC-3126-SM-5GZZ2 146.90             46.09              41.84           
GTEX-111FC-3326-SM-5GZYV 134.00             51.58              35.34           
GTEX-111YS-0006-SM-5NQBE  29.03             19.64              16.53           
GTEX-1122O-0003-SM-5Q5DL 120.40             48.86             102.40           
GTEX-1122O-0005-SM-5O99J

In [8]:
gtex_v8_target <- gtex_v8_target |> dplyr::filter(SAMPID %in% common_samples)
head(gtex_v8_target)

SAMPID                   SMATSSCR SMCENTER SMPTHNTS SMRIN SMTS 
1 GTEX-1117F-3226-SM-5N9CT  1       B1       2 pieces  6.2  Brain
2 GTEX-111FC-3126-SM-5GZZ2  1       B1       2 pieces  6.1  Brain
3 GTEX-111FC-3326-SM-5GZYV  2       B1       2 pieces  7.1  Brain
4 GTEX-111YS-0006-SM-5NQBE NA       B1       NA        8.2  Blood
5 GTEX-1122O-0003-SM-5Q5DL NA       B1       NA       10.0  Blood
6 GTEX-1122O-0005-SM-5O99J NA       B1       NA        7.6  Blood
  SMTSD                               SMUBRID     SMTSISCH SMTSPAX ⋯ SME1PCTS
1 Brain - Cortex                      0001870     1418     931     ⋯ 50.4180 
2 Brain - Cortex                      0001870     1221     697     ⋯ 50.5361 
3 Brain - Cerebellum                  0002037     1221     697     ⋯ 49.9638 
4 Whole Blood                         0013756     -121      NA     ⋯ 49.8542 
5 Cells - EBV-transformed lymphocytes EFO_0000572  -77      NA     ⋯ 50.0182 
6 Whole Blood                         0013756      -77      NA     ⋯ 49.9416 
  SMRRNART   SME1MPRT SMNUM5CD SMDPMPRT SME2PCTS SUBJID     SEX AGE   DTHHRDY
1 0.06203630 0.993989 NA       0        50.1910  GTEX-1117F 2   60-69 4      
2 0.03638110 0.995651 NA       0        50.0212  GTEX-111FC 1   60-69 1      
3 0.01607660 0.994173 NA       0        50.1835  GTEX-111FC 1   60-69 1      
4 0.00278633 0.988644 NA       0        50.3820  GTEX-111YS 1   60-69 0      
5 0.00484995 0.995954 NA       0        50.1247  GTEX-1122O 2   60-69 0      
6 0.00286035 0.995023 NA       0        50.3328  GTEX-1122O 2   60-69 0

Check the sample counts falling under each category

In [9]:
table(gtex_v8_target$SMTS)


Blood Brain 
  929  2642 

### Load Gene id to symbol mapping
Will be used later to map ensembl id back to symbols for better interpretation

In [10]:
gtex_v8_gene_id_to_symbol <- read_csv("gtex_v8_gene_id_to_symbol.csv")
head(gtex_v8_gene_id_to_symbol)

Rows: 5000 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (2): Name, Symbol

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Name               Symbol 
1 ENSG00000244734.3  HBB    
2 ENSG00000188536.12 HBA2   
3 ENSG00000210082.2  MT-RNR2
4 ENSG00000198804.2  MT-CO1 
5 ENSG00000198712.1  MT-CO2 
6 ENSG00000198938.2  MT-CO3

## Regression

### Multiple variable Linear regression of tissue on gene expression

In [11]:
Y_scale <- scale(Y[gtex_v8_target$SAMPID, ])

# Perform regression
results <- lapply(1:ncol(Y_scale), function(i) {
  gene_expression <- Y_scale[,i]
  model <- lm(gene_expression ~ SMTS + AGE + SEX, data = gtex_v8_target)
  summary(model)
})

In [12]:
brain_coefficients <- data.frame()

for (i in 1:length(results)) {
  coef_matrix <- coef(results[[i]])
  if ("SMTSBrain" %in% rownames(coef_matrix)) {
    coef_row <- as.data.frame(coef(results[[i]]))['SMTSBrain',]
    gene_name <- colnames(Y_scale)[i] # gene ensembl id
    coef_row$Name <- gene_name
    brain_coefficients <- rbind(brain_coefficients, coef_row)
  }
}

rownames(brain_coefficients) <- NULL
head(brain_coefficients)

Estimate   Std. Error t value    Pr(>|t|)      Name              
1  0.2557212 0.03919994   6.523509  7.836851e-11 ENSG00000168137.15
2  0.8067258 0.03696221  21.825693  3.185476e-99 ENSG00000144550.12
3  1.1231343 0.03446424  32.588399 4.053116e-204 ENSG00000134072.10
4 -0.6253443 0.03607984 -17.332233  1.125507e-64 ENSG00000171148.13
5 -1.6771324 0.02466063 -68.008504  0.000000e+00 ENSG00000241553.12
6  1.0569137 0.03508128  30.127568 7.811931e-178 ENSG00000163703.17

In [13]:
top_DEGs <- brain_coefficients |>
  dplyr::arrange(desc(abs(`t value`))) |>
  head(5) |>
  dplyr::left_join(gtex_v8_gene_id_to_symbol, by = "Name")
top_DEGs

Estimate  Std. Error t value   Pr(>|t|) Name               Symbol 
1 -2.026069 0.01772878 -114.2814 0        ENSG00000104894.11 CD37   
2 -2.004640 0.01785012 -112.3040 0        ENSG00000128340.14 RAC2   
3 -1.972891 0.01853225 -106.4572 0        ENSG00000162511.7  LAPTM5 
4  1.972961 0.01877580  105.0800 0        ENSG00000198899.2  MT-ATP6
5 -1.950907 0.01903777 -102.4756 0        ENSG00000122122.9  SASH3

The positive estimated coefficient of around 2 for MT-ATP6 suggests it is more highly expressed in brain tissue compared to in blood (roughly two standard deviations over the expression in blood on average). It made senset hat MT-ATP6 is expressed more in brain because brain consumes a lot of energy and have more mitochondrions.

The negative coefficients estimates of around 2 for UBA52, LAPTM5, UBA52, RAC2 and CD37 indicate higher expression in blood (expression in brain is roughly two standard deviations below blood on average). It made sense that CD37 is expressed more in blood as it is a Leukocyte antigen.



### Logistic regression predicting sex from gene expression

In [14]:
subset_gtex_v8_target <- gtex_v8_target |>
  dplyr::filter(SMTS == "Blood") |>
  dplyr::mutate(SEX = ifelse(SEX == 2, 1, 0))

# Scale the gene expression data for the subset
Y_scale <- scale(Y[subset_gtex_v8_target$SAMPID, ])

# Perform logistic regression for each gene
results <- lapply(1:ncol(Y_scale), function(i) {
  gene_expression <- Y_scale[, i]
  model <- glm(SEX ~ gene_expression, data = subset_gtex_v8_target, family = binomial)
  summary(model)
})

Warning message:
“glm.fit: fitted probabilities numerically 0 or 1 occurred”
Warning message:
“glm.fit: algorithm did not converge”
Warning message:
“glm.fit: fitted probabilities numerically 0 or 1 occurred”
Warning message:
“glm.fit: fitted probabilities numerically 0 or 1 occurred”
Warning message:
“glm.fit: fitted probabilities numerically 0 or 1 occurred”
Warning message:
“glm.fit: fitted probabilities numerically 0 or 1 occurred”


In [15]:
coef_matrix <- coef(results[[1]])
coef_row <- as.data.frame(coef(results[[i]]))["gene_expression", ]

In [18]:
sex_coefficients <- data.frame()

for (i in 1:length(results)) {
  coef_matrix <- coef(results[[i]])
  if ("gene_expression" %in% rownames(coef_matrix)) {
    coef_row <- as.data.frame(coef(results[[i]]))["gene_expression", ]
    coef_row[,1:2] = exp(coef_row[,1:2])
    gene_name <- colnames(Y_scale)[i] # gene ensembl id
    coef_row$Name <- gene_name
    sex_coefficients <- rbind(sex_coefficients, coef_row)
  }
}
rownames(sex_coefficients) <- NULL
head(sex_coefficients)

Estimate  Std. Error z value     Pr(>|z|)  Name              
1 1.0287194 1.071302    0.41110452 0.6809959 ENSG00000168137.15
2 0.9802623 1.072168   -0.28608453 0.7748134 ENSG00000144550.12
3 0.8902385 1.075181   -1.60390247 0.1087356 ENSG00000134072.10
4 1.0658108 1.071255    0.92598060 0.3544561 ENSG00000171148.13
5 1.0618048 1.071377    0.86983394 0.3843912 ENSG00000241553.12
6 1.0026704 1.071708    0.03850868 0.9692821 ENSG00000163703.17

In [19]:
top_DEGs <- sex_coefficients |>
  dplyr::arrange(desc(abs(`z value`))) |>
  head(5) |>
  dplyr::left_join(gtex_v8_gene_id_to_symbol, by = "Name")
top_DEGs

Estimate      Std. Error   z value   Pr(>|z|)     Name               Symbol
1 4.155169e+159 2.376347e+17  9.186200 4.069629e-20 ENSG00000229807.10 XIST  
2  1.822809e+00 1.077373e+00  8.056000 7.883178e-16 ENSG00000126012.11 KDM5C 
3  1.623546e+00 1.073216e+00  6.858359 6.965598e-12 ENSG00000198034.10 RPS4X 
4  1.595803e+00 1.073604e+00  6.580825 4.678460e-11 ENSG00000215301.9  DDX3X 
5  4.019652e-98 1.313242e+16 -6.042544 1.517028e-09 ENSG00000067048.16 DDX3Y

High magnitude (>>2) of the exponentiated coefficient estimate of XIST relative to the other genes with positive coefficient estimates indicate that its over expression is very strongly associated with the female sex without correcting for any other covariatess with a crazy big odds ratio. The smaller magnitude of exponentiated coefficient estaimates for KDM5C, RPS4X, DDX3X, UBA1 of ranging from 1.5 to 1.8 represent a 50 to 80% increase odds of being female comparing individuals who differ in 1 sd of gene expression, correcting for no other covariates. These results make sense as XIST, KDM5C, RPS4X, DDX3X, UBA1 are located on the X chromosome so females might have more expression of these genes provided that they are less impacted by X-chromosome deactivation.

The close to zero of the exponentiated coefficient estimate for DDX3Y suggest that the the high expression of this gene is strongly negatively associated with being female. This made sense as DDX3Y is on Y chromosome and expressed only in male, and every 1 sd increase in expression of DDX3Y is asscoiated with almost 100% decreased odds of being a female.